### PR TITLE
Rework usage of pyupgrade

### LINKED ISSUE #167: Rework usage of pyupgrade
## Summary
Rework the usage of the [pyupgrade](https://github.com/asottile/pyupgrade) tool, to be part of the code style (black, isort, ...) checks and verification.

## Tasks
- [x] Remove pyupgrade from precomit hooks
- [x] Add pyupgrade target to `noxfile.py` 
- [x] Add pyupgrade target to be part of the following targets verify/check/fix

### PR DESCRIPTION
**Pull request by Nicoretti**
fixes #167 

- Add pyupgrade as dev dependency
- Add pyupgrade task(s) to noxfile
- Add pyupgrade to verification step
- Update commit hooks to use nox targets

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -123,10 +130,27 @@ def fix(session: Session) -> None:
...
+    session.run(
+        "poetry",
+        "run",
+        "python",
+        "-m",
+        "pyupgrade",
+        "--py38-plus",
+        "--exit-zero-even-if-changed",
+        *files,
+    )
```
Comment (tkilias) [inline noxfile.py]: why do you calling pyuprdae here directly and have further down a session

Comment (tkilias) [reply noxfile.py]: ah, I see it --exit-zero-even-if-changed, so one is for precommit and one for CI or local, maybe explain this a bit or have to session for pyupgrade where the name explains it

**Review by tkilias — CHANGES_REQUESTED** [review]

File: noxfile.py (hunk 1 above)
Comment (Nicoretti) [reply noxfile.py]: Isn't it clear enough from the fact that it is part of the "fix" session? 
What would be best in your opinion `comment`?

Comment (tkilias) [reply noxfile.py]: maybe move it also to a session with the name pyupgrade_fix or fix_pyupgrade and call this session.

Comment (Nicoretti) [reply noxfile.py]: Hi @tkilias, thanks for clarifying, still I personally rather would keep it as is, reasoning:
* overhead is not worth it in my opinion
* consistency with the other call's within the fix action

anyway I will go with whatever your final call will be.

Comment (tkilias) [reply noxfile.py]: Third option, I think, my problem was understanding that this call is actually  configured for fix. Maybe, it is enough to extract it to a method

**Review by tkilias — APPROVED** [review]

**Pull request merged**